nikola/log.py

```diff
@@ -137,6 +136,39 @@ def get_logger(name: str, handlers=None) -> logging.Logger:
 
 
 LOGGER = get_logger("Nikola")
+TEMPLATES_LOGGER = get_logger("nikola.templates")
+
+
+def init_template_trace_logging(filename: str) -> None:
+    """Initialize the tracing of the template system.
+
+    This tells a theme designer which templates are being exercised
+    and for which output files, and, if applicable, input files.
+
+    As there is lots of other stuff happening on the normal output stream,
+    this info is also written to a log file.
+    """
+    TEMPLATES_LOGGER.level = logging.DEBUG
+    formatter = logging.Formatter(
+        fmt=_LOGGING_FMT,
+        datefmt=_LOGGING_DATEFMT,
+    )
+    shandler = logging.StreamHandler()
+    shandler.setFormatter(formatter)
+    shandler.setLevel(logging.DEBUG)
+
+    fhandler = logging.FileHandler(filename, encoding="UTF-8")
+    fhandler.setFormatter(formatter)
+    fhandler.setLevel(logging.DEBUG)
+
+    TEMPLATES_LOGGER.handlers = [shandler, fhandler]
+    TEMPLATES_LOGGER.propagate = False
+
+    TEMPLATES_LOGGER.info("Template usage being traced to file %s", filename)
+
+
+if TEMPLATES_TRACE:
+    init_template_trace_logging("templates_log.txt")
```

Comment:
Consider using the `.log` extension, to avoid confusion with things like reST posts (which can also end with `.txt`).

```suggestion
    init_template_trace_logging("templates_trace.log")
```